lib/cfa/matcher.rb

```diff
@@ -41,4 +75,14 @@ def value_match?(element, matcher)
       end
     end
   end
+
+  # An abstract class documenting the block argument to {Matcher#initialize}
+  class MatcherBinaryPredicate < Proc
```

Comment:
Again I do not like it as it affect production code only for documentation purpose.